crates/sovereign-sdk/rollup-interface/src/state_machine/stf.rs

```diff
@@ -273,6 +293,48 @@ pub trait StateTransitionFunction<Vm: Zkvm, Da: DaSpec> {
     ) -> (Self::StateRoot, CumulativeStateDiff, SpecId);
 }
 
+#[derive(Debug)]
+/// Error that can occur during appyling a soft confirmation
+pub enum SoftConfirmationError {
+    /// The public key of the sequencer (known by a full node or prover) does not match
+    /// the public key in the soft confirmation
+    SequencerPublicKeyMismatch,
+    /// The DA hash in the soft confirmation does not match the hash of the DA block provided
+    InvalidDaHash,
+    /// The DA tx commitment in the soft confirmation does not match the tx commitment of the DA block provided
+    InvalidDaTxsCommitment,
+    /// The hash of the soft confirmation is incorrect
+    InvalidSoftConfirmationHash,
+    /// The soft confirmation signature is incorret
+    InvalidSoftConfirmationSignature,
+    /// Any other error that can occur during the application of a soft confirmation
+    /// These can come from runtime hooks etc.
+    Other(String),
```

Comment:
This will fail no_std

Reply:
https://doc.rust-lang.org/stable/alloc/string/struct.String.html